mlrun/model_monitoring/db/tsdb/preaggregate.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Optional
+
+import mlrun.errors
+import mlrun.utils
+
+
+@dataclass
+class PreAggregateConfig:
+    """Configuration for pre-aggregated tables and retention policies."""
+
+    aggregate_intervals: list[str] = None
+    agg_functions: list[str] = None
+    retention_policy: dict[str, str] = None
+
+    def __post_init__(self):
+        if self.aggregate_intervals is None:
+            self.aggregate_intervals = ["10m", "1h", "6h", "1d", "1w", "1M"]
+
+        if self.agg_functions is None:
+            self.agg_functions = ["sum", "avg", "min", "max", "count", "last"]
+
+        if self.retention_policy is None:
+            self.retention_policy = {
+                "raw": "7d",
+                "10m": "30d",
+                "1h": "1y",
+                "6h": "1y",
+                "1d": "5y",
+                "1w": "5y",
+                "1M": "5y",
+            }
+
+
+class PreAggregateHandler:
+    """Handles pre-aggregate validation, time alignment, and optimization decisions."""
+
+    def __init__(self, pre_aggregate_config: Optional[PreAggregateConfig] = None):
+        """
+        Initialize the pre-aggregate handler.
+
+        :param pre_aggregate_config: Configuration for pre-aggregated tables and operations.
+                                   If None, all pre-aggregate operations will be disabled.
```

Comment:
Indentation and mention of the default.
```suggestion
        If None (the default), all pre-aggregate operations will be disabled.
```

Reply:
will fix